paasta_tools/cli/utils.py

```diff
@@ -731,6 +731,27 @@ def validate_given_deploy_groups(all_deploy_groups, args_deploy_groups):
     return valid_deploy_groups, invalid_deploy_groups
 
 
+def short_to_full_git_sha(short, refs):
+    """Converts a short git sha to a full sha
+
+    :param short: A short git sha represented as a string
+    :param refs: A list of refs in the git repository
+    :return: The full git sha or None if one can't be found
+    """
+    for sha in list(set(refs.values())):
+        if sha.startswith(short):
+            return sha
+    return None
```

Comment:
```python
return [sha for sha in set(refs.values()) if sha.startswith(short)]
```

howcome you're doing list(set(... ? to remove dupes? if so, you can leave it as a set and just iterate through that.